core/src/main/scala/org/bitcoins/core/protocol/Address.scala

```diff
@@ -96,8 +98,8 @@ sealed abstract class Bech32Address extends BitcoinAddress {
 }
 
 object Bech32Address extends AddressFactory[Bech32Address] {
-  private case class Bech32AddressImpl(hrp: HumanReadablePart, data: Vector[UInt8]) extends Bech32Address {
-    verifyChecksum(hrp, UInt8.toBytes(data))
+  private case class Bech32AddressImpl(hrp: HumanReadablePart, data: Vector[UInt5]) extends Bech32Address {
+    //require(verifyChecksum(hrp, data), "checksum did not pass")
```

Comment:
There is a weird thing here that I can't decide on, should the `checksum` be included in `data`? Currently we provide `data` to `createChecksum` to actually make the checksum. If we have the `checksum` embedded in `data` we need to remove it from `data` before we call `verifyChecksum`. 

Since we do not have the `checksum` inside of `Vector[UInt5]` here we cannot check it.

The checksum IS checked [here](https://github.com/bitcoin-s/bitcoin-s-core/blob/master/core/src/main/scala/org/bitcoins/core/protocol/Address.scala#L240) when calling `Bech32Address.fromString`, but we should probably put the invariant in the case class itself so we can never have a `Bech32AddressImpl` created without a valid checksum. 

Also note in the old diff that there was no `require()` around the `verifyChecksum`, so this check was useless before.

That's a long winded way of saying, yes we should do it. I don't think we should do it on this PR though